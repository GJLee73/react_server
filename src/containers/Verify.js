/**
 * 증명 페이지. 기존 이메일에 보내는 버튼과 새로 만든 이메일에 보내는 버튼이 존재.
 * @author G1
 * @logs // 18.2.25
 */

import React, { Component } from 'react';
import {VerifyComponent} from '../components/auth'
import {connect} from 'react-redux'
import {reSendEmail} from '../modules/authentication'
import {emailRequest} from '../modules/register'
//test를 위한 export선언.
export class Verify extends Component {
    constructor(props) {
        super(props);
        this.state = { user:'' }
        this.handleReSend=this.handleReSend.bind(this)
        this.checkEmail=this.checkEmail.bind(this)
    }

    componentWillMount(){
            console.log(3)
            this.setState({user:this.props.user})
            console.log(this.props.user)
					// this.props.cleanCurrentUser()
    }
    /**
     * 이메일로 증명 링크를 보낸다.
     * @param {string} email 
     * @description
     * 링크를 보낸 후 로그인 페이지로 리다이렉트 시킨다.
     */
    handleReSend(email){
        
        this.props.reSendEmail(email,this.state.user).then(()=>{
            alert('메시지를 보냈습니다!')
            this.props.history.push('/login')
        })
           
        
    }
    /**
     * 새로운 이메일이 중복되는지 확인한다.
     * @param {string} email - 이메일
     */
    checkEmail(email){
		console.log(email)
		this.props.emailRequest(email).then(()=>{
            this.handleReSend(email)
		}).catch(()=>{
            alert('이메일이 중복됩니다.')
        })
    }

    render() { 
        return ( 
        <div>
            <VerifyComponent verify={this.handleReSend} checkEmail={this.checkEmail}/>
        </div> )
    }
}
 
const mapStateToProps = (state) => {
	return {
		user:state.authentication.getIn(['status','currentUser'])
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		reSendEmail: (id, pw)=>{
			return dispatch(reSendEmail(id,pw));
		},emailRequest:(email)=>{
            return dispatch(emailRequest(email))
        },
        // cleanCurrentUser:()=>{
        //     return dispatch(cleanCurrentUser())
        // }
	};
};


export default connect(mapStateToProps, mapDispatchToProps)(Verify);