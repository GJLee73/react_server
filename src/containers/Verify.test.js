import React from 'react'
import {mount,shallow} from 'enzyme'
import {Verify} from './Verify'

describe('verify test',()=>{
    var cleanCurrentUser = jest.fn()
    var reSendEmail = jest.fn()
    it('성공적으로 렌더링되어야 합니다.', () => {
        shallow(<Verify cleanCurrentUser={cleanCurrentUser} reSendEmail={reSendEmail}/>);
       
      });
     //reactrouter가 포함되어 있으면 스냅샷 테스트 불가. 따라서 다음과 같이 initialEntreis를 정해준다.
      it('snapshot', () => {
        const wrapper = mount(<Verify cleanCurrentUser={cleanCurrentUser} reSendEmail={reSendEmail}/> )
        expect(wrapper).toMatchSnapshot();
      })
      it('fn called',()=>{
        const wrapper = shallow(<Verify cleanCurrentUser={cleanCurrentUser} reSendEmail={reSendEmail}/>);
        expect(wrapper)
      })
})