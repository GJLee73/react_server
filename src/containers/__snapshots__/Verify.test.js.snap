// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify test renders correctly 1`] = `
<Verify
  cleanCurrentUser={[Function]}
  reSendEmail={[Function]}
>
  <div>
    <VerifyComponent
      checkEmail={[Function]}
      verify={[Function]}
    >
      <div>
        <div
          className="container auth"
        >
          <p>
            인증되지 않은 아이디 입니다. 이메일을 확인해 주세요!!!! 아니면 이메일을 재 전송해 주세요
          </p>
          <div
            className="card-content"
          >
            <div
              className="row"
            >
              <a
                className="waves-effect waves-light btn"
                onClick={[Function]}
              >
                기존 이메일에 보내기
              </a>
              <hr />
              <div
                className="input-field col s12"
              >
                <label>
                   이메일 새로 작성하기
                </label>
                <input
                  className="validate"
                  name="email"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <a
                className="waves-effect waves-light btn"
                onClick={[Function]}
              >
                SUBMIT
              </a>
            </div>
          </div>
        </div>
      </div>
    </VerifyComponent>
  </div>
</Verify>
`;

exports[`verify test snapshot 1`] = `
<Verify
  cleanCurrentUser={[Function]}
  reSendEmail={[Function]}
>
  <div>
    <VerifyComponent
      checkEmail={[Function]}
      verify={[Function]}
    >
      <div>
        <div
          className="container auth"
        >
          <p>
            인증되지 않은 아이디 입니다. 이메일을 확인해 주세요!!!! 아니면 이메일을 재 전송해 주세요
          </p>
          <div
            className="card-content"
          >
            <div
              className="row"
            >
              <a
                className="waves-effect waves-light btn"
                onClick={[Function]}
              >
                기존 이메일에 보내기
              </a>
              <hr />
              <div
                className="input-field col s12"
              >
                <label>
                   이메일 새로 작성하기
                </label>
                <input
                  className="validate"
                  name="email"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <a
                className="waves-effect waves-light btn"
                onClick={[Function]}
              >
                SUBMIT
              </a>
            </div>
          </div>
        </div>
      </div>
    </VerifyComponent>
  </div>
</Verify>
`;
